File: scraper.js
const Chrome = require('./lib/chrome');
const sleep = require('./lib/sleep.js');
const fs = require('fs');
const path = require('path');

(async () => {
  // initialize browser
  const chrome = new Chrome("http://example.com", 0, 0, 800, 600, true);
  await chrome.start();
  await chrome.goto("https://shopee.tw/a-i.178926468.21448123549")
  await sleep(5000);
  await chrome.runScript("window.scrollTo(0, document.body.scrollHeight)")
  await sleep(5000);
  const htmlContent = await chrome.runScript("document.body.outerHTML");
  
  // create 'result' directory if it doesn't exist
  const resultDir = path.join(__dirname, 'result');
  if (!fs.existsSync(resultDir)) {
    fs.mkdirSync(resultDir);
  }
  
  // generate a unique filename using the timestamp
  const filename = `page_${Date.now()}.html`;
  const filepath = path.join(resultDir, filename);

  // save the HTML content to the file
  fs.writeFileSync(filepath, htmlContent);
  console.log(`Saved to ${filepath}`);
  
  chrome.quit();
  process.exit(0);
})();

File: scraper.py
import asyncio
import os
from pathlib import Path
from random import randint

from loguru import logger

from lib.chrome import Chrome

CUR_DIR = str(Path(__file__).parent.absolute())
TEMP_DIR = os.path.join(CUR_DIR, "temp")
OUTPUT_DIR = os.path.join(CUR_DIR, "output")
DONE_MARKER_NAME = "done"


async def work():
    try:
        chrome = None
        user_data_dir = os.path.join(TEMP_DIR, "profile")

        chrome = Chrome(
            width=800 + randint(0, 200),
            height=600 + randint(0, 100),
            block_image=True,
            user_data_dir=user_data_dir,
        )
        await chrome.start()
        await chrome.goto("google.com")
        await asyncio.sleep(5)
        await chrome.run_script("window.scrollTo(0, document.body.scrollHeight);")

    except Exception as ex:
        logger.exception(ex)
    finally:
        await chrome.quit()

async def main():
    await work()


if __name__ == "__main__":
    asyncio.run(main())
